tests/virt_autotest/setup_dns_service.pm

```diff
@@ -0,0 +1,202 @@
+\; ?
+\\\$ORIGIN $dns_domain_reverse.in-addr.arpa.
+\\\$TTL 1D
+\@          IN SOA ns.$dns_domain_forward. mailmaster.$dns_domain_forward. (
+              \$(date \\+\\\%Y\\\%m\\\%d)        ; serial
+              21600             ; refresh
+              3600              ; retry
+              3600000           ; expire
+              86400 )           ; minimum
+
+
+\; ----------- ENREGISTREMENTS -----------
+\@                       IN NS                   ns.$dns_domain_forward.
+\;\;
+\$(echo -e $dns_records_reverse)
+\;\;
+\; ----------- ENREGISTREMENTS SPECIAUX -----------";
+    my $set_dns_reverse_zone = "cat > $dns_reverse_zone_file \<\<EOF\n$dns_reverse_zone\nEOF\n";
+    type_string("$set_dns_reverse_zone");
+
+    my $dns_config_file      = "/etc/named.conf";
+    my $dns_config_file_tmp  = "/etc/named.conf.tmp";
+    my $dns_service_name     = "named";
+
+    assert_script_run("cp $dns_config_file $dns_config_file.orig");
+
+    #The following forward zone info will be inserted into /etc/named.conf
+    #zone "testvirt.net" in {
+    #        type master;
+    #        file "testvirt.net.zone";
+    #};
+    my $set_dns_config_zone = "awk \'done != 1 && /^zone.*\$/ { print \"zone \\\"$dns_domain_forward\\\" in {\\n        type master;\\n        file \\\"$dns_domain_forward\.zone\\\";\\n};\"; done=1 } 1\' $dns_config_file > $dns_config_file_tmp";
+    assert_script_run("$set_dns_config_zone");
+    assert_script_run("mv $dns_config_file_tmp $dns_config_file");
+
+    #The following reverse zone info will be inserted into /etc/named.conf
+    #zone "123.168.192.in-addr.arpa" in {
+    #        type master;
+    #        file "123.168.192.zone";
+    #};
+    $set_dns_config_zone = "awk \'done != 1 && /^zone\.*\$/ { print \"zone \\\"$dns_domain_reverse\.in-addr\.arpa\\\" in {\\n        type master;\\n        file \\\"$dns_domain_reverse\.zone\\\";\\n};\"; done=1 } 1\' $dns_config_file > $dns_config_file_tmp";
+    assert_script_run("$set_dns_config_zone");
+    assert_script_run("mv $dns_config_file_tmp $dns_config_file");
+
+    my $get_os_installed_release = "lsb_release -r | grep -oE \"[[:digit:]]{2}\"";
+    my $os_installed_release     = script_output($get_os_installed_release, $wait_script, type_command => 0, proceed_on_failure => 0);
+    if ($os_installed_release gt '11') {
+        assert_script_run("systemctl enable $dns_service_name");
+        assert_script_run("systemctl restart $dns_service_name");
+        assert_script_run("systemctl status $dns_service_name");
+    }
+    else {
+        assert_script_run("service $dns_service_name restart");
+        assert_script_run("service $dns_service_name status");
+    }
+
+    diag("SSH Connection to virtual machines by using DNS names is working now !");
+
+    foreach (@vm_hostnames_array)
+    {
+	assert_script_run("virsh destroy $_");
+    }
+
+    upload_logs("$dns_config_file.orig");
+    upload_logs("$dns_config_file");
```

Comment:
maybe also upload zone files can help on some connection failures?

Reply:
All logs will not be uploaded in the main run function. They will be printed out in post_fail_hook if failure happens.